### PR TITLE
Atom 5013 add suspend restore actions

### PR DESCRIPTION
**Pull request by dougperez**

### CODE REVIEW

File: lib/conduit/cogotech/actions/change_number.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'conduit/cogotech/actions/base'
+
+module Conduit::Driver::Cogotech
+  class ChangeNumber < Conduit::Driver::Cogotech::Base
+    required_attributes :meid, :new_mdn
```
Comment (pedromartinez) [inline lib/conduit/cogotech/actions/change_number.rb]: new_mdn? This let's you pick the mdn?

File: lib/conduit/cogotech/decorators/change_number.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require_relative 'base'
+
+module Conduit::Cogotech::Decorators
+  class ChangeNumber < Base
+    def change_number_attributes
```
Comment (pedromartinez) [inline lib/conduit/cogotech/decorators/change_number.rb]: Any reason not to just bake this into the view

File: lib/conduit/cogotech/actions/change_number.rb (hunk 1 above)
Comment (dougperez) [inline lib/conduit/cogotech/actions/change_number.rb]: Yeah, http://docs.cogotech.apiary.io/#reference/device-apis/device-cruds/swap-device-mdn.
They do have a change_did action. http://docs.cogotech.apiary.io/#reference/device-apis/device-cruds/change-device-did, that returns a new mdn.

Comment (pedromartinez) [inline lib/conduit/cogotech/actions/change_number.rb]: Ah okay, well the "change-device-did" maps more to the standard "change_number" of other carriers, FYI.

Comment (dougperez) [inline lib/conduit/cogotech/actions/change_number.rb]: Yeah, any preference on what to call this action?

File: lib/conduit/cogotech/decorators/change_number.rb (hunk 2 above)
Comment (dougperez) [inline lib/conduit/cogotech/decorators/change_number.rb]: Hmm, no this could be baked into the view. Are you asking for this one instance or all of them?

Comment (pedromartinez) [inline lib/conduit/cogotech/decorators/change_number.rb]: I guess I'm wondering why we're using a decorator for something that seems redundant with the view?

File: lib/conduit/cogotech/actions/change_number.rb (hunk 1 above)
Comment (pedromartinez) [inline lib/conduit/cogotech/actions/change_number.rb]: Hmm, maybe change_device_mdn like they have it?

File: lib/conduit/cogotech/decorators/change_number.rb (hunk 2 above)
Comment (dougperez) [inline lib/conduit/cogotech/decorators/change_number.rb]: No reason yet, just stayed consistent between the other drivers. I have not found a reason to need it yet like we do for Sprint.

File: lib/conduit/cogotech/actions/base.rb [hunk 3]
```diff
@@ -18,11 +18,15 @@ def remote_url
     end
 
     def view_context
```
Comment (pedromartinez) [inline lib/conduit/cogotech/actions/base.rb]: I'd probably get rid of this method and work it into the view, if possible.

File: lib/conduit/cogotech/actions/change_device.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+module Conduit::Driver::Cogotech
+  class ChangeDevice < Conduit::Driver::Cogotech::Base
+    required_attributes :meid, :new_meid
+    http_method         :post
+
+    private
+
+    def action_url
+      "/#{@options[:meid]}/SwapDeviceID?requestId=#{@options[:request_id]}"
```
Comment (pedromartinez) [inline lib/conduit/cogotech/actions/change_device.rb]: I think request_id should just be a method on the action base class. It's just supposed to be a uuid generated on our side, I think.

File: lib/conduit/cogotech/actions/base.rb (hunk 3 above)
Comment (dougperez) [inline lib/conduit/cogotech/actions/base.rb]: Yeah we dont need it anymore. Removed.

File: lib/conduit/cogotech/parsers/change_device_mdn.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require 'conduit/cogotech/parsers/base'
+
+module Conduit::Driver::Cogotech
+  class ChangeDeviceMdn::Parser < Parser::Base
+    attribute :transaction_id
+
+    def response_content?
+      !@json.nil?
```
Comment (pedromartinez) [inline lib/conduit/cogotech/parsers/change_device_mdn.rb]: Is this the same for all of them? Thoughts about moving it to Parser::Base?

File: lib/conduit/cogotech/actions/change_device.rb (hunk 4 above)
Comment (dougperez) [inline lib/conduit/cogotech/actions/change_device.rb]: Yeah I wasn't sure what it was either. I was going to polish that up on another pull request pass once these actions gets merged.

File: lib/conduit/cogotech/parsers/change_device_mdn.rb (hunk 5 above)
Comment (dougperez) [inline lib/conduit/cogotech/parsers/change_device_mdn.rb]: I like it, moved to base and removed.

**Comment by pedromartinez** [comment]
👍

**Pull request merged**